aten/src/ATen/core/LazyOps.cpp

```diff
@@ -0,0 +1,28 @@
+#include <ATen/core/LazyTensor.h>
+
+namespace at {
+
+bool Tensor::is_lazy() const {
+  return impl_->is_lazy();
+}
```

Comment:
It is a little funny for a Tensor method to be defined all the way over here, rather than in `Tensor.cpp` where most people would expect it to be defined

Reply:
Actually, it's worse than this. You define `is_lazy` in `native_functions.yaml` which means the codegen should be defining its own copy of `Tensor::is_lazy`, so now you have two definitions of `is_lazy`. Two definitions enter a linker, only one emerges... only the linker knows who...

Reply:
Is there a good reason for `is_lazy` to be in `native_functions.yaml`; in particular, do we really want `Tensor.is_lazy` to exist in Python? (If you asked @zdevito, he'd say no!)

Reply:
(A further aside: there isn't a good way to write functions in `native_functions.yaml` that bypass dispatch entirely and go straight to TensorImpl, as you do here. All of the functions like this are special cased in some way or another...)

Reply:
>Is there a good reason for is_lazy to be in native_functions.yaml

I'd like `is_lazy` to exist in Python because we need to avoid some size checks.  E.g. in `torch/autograd/__init__.py` but further in operators like batch norm.

>A further aside: there isn't a good way to write functions in native_functions.yaml that bypass dispatch entirely

Should I just use dispatch then?

Reply:
Anything is fine; this test isn't perf critical so it's not a big deal if it goes through the dispatcher.